server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/codeSearch.ts

```diff
@@ -0,0 +1,204 @@
+            this.logging.info(`Code searched succeed with num of results: "${sanitizedSearchResults.length}"`)
+            return sanitizedSearchResults
+        } catch (error: any) {
+            this.logging.error(`Error during code search: ${error.message || error}`)
+            throw error
+        }
+    }
+
+    /**
+     * Parses chunks from vector index search into CodeSearchOutput format
+     * Based on the queryRelevantDocuments method pattern
+     */
+    private parseChunksToCodeSearchOutput(chunks: Chunk[]): CodeSearchOutput[] {
+        const codeSearchResults: CodeSearchOutput[] = []
+        if (!chunks) {
+            return codeSearchResults
+        }
+
+        for (const chunk of chunks) {
+            // Extract content and context
+            const text = chunk.content || ''
+            const relativeFilePath = chunk.relativePath ?? path.basename(chunk.filePath)
+
+            // Extract line information
+            const startLine = chunk.startLine ?? -1
+            const endLine = chunk.endLine ?? -1
+
+            // Create the base search result
+            const baseSearchResult = {
+                text,
+                relativeFilePath,
+                startLine,
+                endLine,
+            }
+
+            // Add programming language information if available
+            if (chunk.programmingLanguage && chunk.programmingLanguage !== 'unknown') {
+                codeSearchResults.push({
+                    ...baseSearchResult,
+                    programmingLanguage: {
+                        languageName: chunk.programmingLanguage,
+                    },
+                })
+            } else {
+                codeSearchResults.push(baseSearchResult)
+            }
+        }
+
+        return codeSearchResults
+    }
+
+    private createOutput(content: string | any[]): InvokeOutput {
+        if (typeof content === 'string') {
+            return {
+                output: {
+                    kind: 'text',
+                    content: content,
+                },
+            }
+        } else {
+            return {
+                output: {
+                    kind: 'json',
+                    content: content,
+                },
+            }
+        }
+    }
+
+    public getSpec() {
+        return {
+            name: 'codeSearch',
+            description:
+                "Find snippets of code from the codebase most relevant to the search query.\nThis is a semantic search tool, so the query should ask for something semantically matching what is needed.\nIf it makes sense to only search in particular directories, please specify them in the target_directories field.\nUnless there is a clear reason to use your own search query, please just reuse the user's exact query with their wording.\nTheir exact wording/phrasing can often be helpful for the semantic search query. Keeping the same exact question format can also be helpful.",
```

Comment:
`target_directories`?

Reply:
update the description to support code search for current workspace as we localProjectContextController doesn't support search against a target dierectory/subdirectory.